includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -112,6 +112,21 @@ public function handle_deactivate_promotion() {
 		exit;
 	}
 
+	/**
+	 * Check if a user can promote a job. They must have permission to manage job listings and the post type must be a published job_listing.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return bool Returns true if they can promote a job.
+	 */
+	private function can_promote_job( int $post_id ) {
+		if ( 'job_listing' !== get_post_type( $post_id ) || 'publish' !== get_post_status( $post_id ) ) {
```

Comment:
I'm not sure if we should do that here, but maybe we can have another restriction to make sure that the job has a non-empty description (post_content)?

I just noticed that JobTarget fails invisibly when a job has no post content/description.. :thinking:

Reply:
Hm! Good point! I'm thinking that we could validate it and display an error in wpjobmanager.com. Maybe it's a very late check, but I don't think it will be common users trying to promote jobs without titles and descriptions. WDYT?

If it makes sense, I can create an issue for that.

Reply:
Since it's failing silently, I added the issue with high priority: https://github.com/Automattic/wpjobmanager.com/issues/503